### PR TITLE
feat: add animationEnabled to TabView

### PR DESCRIPTION
**Pull request by szymonrybczak**
**Motivation**

According to this issue https://github.com/satya164/react-native-tab-view/issues/1371 I added the `disableChangeTabAnimation` to disable animation when changing tab.

### CODE REVIEW

**Review by okwasniewski — CHANGES_REQUESTED** [review]

File: src/PagerViewAdapter.tsx [hunk 1]
```diff
@@ -72,8 +79,14 @@ export default function PagerViewAdapter<T extends Route>({
...
     if (indexRef.current !== index) {
-      pagerRef.current?.setPage(index);
+      if (animationEnabled) {
+        pagerRef.current?.setPage(index);
+      } else {
+        pagerRef.current?.setPageWithoutAnimation(index);
+        position.setValue(index);
+      }
     }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (satya164) [inline src/PagerViewAdapter.tsx:89]: Don't disable the lint rule. The lint rule is valid - the code in this function will break if `animationEnabled` isn't in the dependency array as it'll only respect what was set initially.

You can instead use `use-latest-callback` package for the `useCallback`, and include the dependency in the dep array for the `useEffect`.

**Review by satya164 — CHANGES_REQUESTED** [review]

**Comment by szymonrybczak** [comment]
@satya164 I updated my code as you said, could you please look at it again?

File: example/package.json [hunk 2]
```diff
@@ -22,7 +22,8 @@
     "react-native": "0.69.5",
     "react-native-pager-view": "5.4.25",
     "react-native-safe-area-context": "4.3.1",
-    "react-native-web": "~0.18.7"
+    "react-native-web": "~0.18.7",
+    "use-latest-callback": "^0.1.5"
```
Comment (okwasniewski) [inline example/package.json:26]: This package is not needed inside `example`.

**Review by satya164 — CHANGES_REQUESTED** [review]
This also needs to be handled in `PanResponderAdapter` for Web, where it should call `panX.setValue` instead of performing an animation

File: src/PagerViewAdapter.tsx (hunk 1 above)
Comment (satya164) [reply src/PagerViewAdapter.tsx:89]: This comment should be removed

**Review by okwasniewski — APPROVED** [review]

File: src/PanResponderAdapter.tsx [hunk 3]
```diff
@@ -81,22 +82,27 @@ export default function PanResponderAdapter<T extends Route>({
...
+            onIndexChangeRef.current(index);
+            pendingIndexRef.current = undefined;
+          }
+        });
+      } else {
+        panX.setValue(offset);
+        onIndexChangeRef.current(index);
+      }
 
       pendingIndexRef.current = index;
```
Comment (satya164) [inline src/PanResponderAdapter.tsx:103]: This should only be set when there is animation, so move it into the `if` block

File: src/PanResponderAdapter.tsx [hunk 4]
```diff
@@ -81,22 +82,27 @@ export default function PanResponderAdapter<T extends Route>({
...
-          toValue: offset,
-          useNativeDriver: false,
-        }),
-      ]).start(({ finished }) => {
-        if (finished) {
-          onIndexChangeRef.current(index);
-          pendingIndexRef.current = undefined;
-        }
-      });
+      if (animationEnabled) {
```
Comment (satya164) [inline src/PanResponderAdapter.tsx:85]: This breaks the swipe gesture. The animation should only be disabled when not swiping similar to how it works on Android & iOS

File: src/types.tsx [hunk 5]
```diff
@@ -55,4 +55,5 @@ export type PagerProps = Omit<
   swipeEnabled?: boolean;
   onSwipeStart?: () => void;
   onSwipeEnd?: () => void;
+  animationEnabled?: boolean;
```
Comment (satya164) [inline src/types.tsx:58]: Move the type next to `swipeEnabled`

File: README.md [hunk 6]
```diff
@@ -339,6 +339,10 @@ Style to apply to the pager view wrapping all the scenes.
 
 Style to apply to the tab view container.
 
+### `animationEnabled`
```
Comment (satya164) [inline README.md:342]: The heading level isn't right, move this to next to `swipeEnabled` and adjust the heading level to match it (`#####`)

**Review by satya164 — CHANGES_REQUESTED** [review]

File: src/PanResponderAdapter.tsx (hunk 4 above)
Comment (szymonrybczak) [reply src/PanResponderAdapter.tsx:85]: Yes, you're right, I didn't noticed that, fixed ✅

**Comment by mlecoq** [comment]
Hi, I have made a separated PR: see #1391 because I need to be able to deactivate animation only for distant tabs (I need to keep them for neighbor tabs)

**Review by satya164 — APPROVED** [review]

**Pull request merged**